src/shared/settingsConfiguration.ts

```diff
@@ -56,4 +59,38 @@ export class DefaultSettingsConfiguration implements SettingsConfiguration {
             return false
         }
     }
+
+    public readDevSetting<T>(key: AwsDevSetting): string
+    public readDevSetting<T>(key: AwsDevSetting, type: string, silent: boolean): T | undefined
+
+    /**
+     * Gets the value of a developer only setting.
+     *
+     * TODO: show a dialog if this is used, and/or make a UI change (such as
+     * changing the AWS Explorer color) so that it's obvious that Toolkit is in
+     * "Developer mode". Throw an error in CI.
+     */
+    public readDevSetting<T>(key: AwsDevSetting, type: string = 'string', silent: boolean = false): T | undefined {
+        const config = vscode.workspace.getConfiguration()
+        const val = config.get<T>(key)
+        if (!val) {
+            const msg = `settings: readDevSetting(): setting "${key}": not found`
+            if (!silent && !isCI()) {
+                throw Error(`AWS Toolkit: ${msg}`)
+            }
+            getLogger().error(msg)
+            return undefined
+        }
+
+        const actualType = typeof val
+        if (actualType !== type) {
+            const msg = `settings: readDevSetting(): setting "${key}": got ${actualType}, expected ${type}`
+            if (!silent) {
```

Comment:
Should this have a similar `!isCI()` statement like the if statement above?

Nit: refactor into a common throw/log/return function.

Reply:
> Should this have a similar `!isCI()` statement like the if statement above?

No this should always fail. If readSetting() returns nonsense then we want a stacktrace instead of  than strange behavior.



> Nit: refactor into a common throw/log/return function.

I tried to think of how to abstract this but don't see anything that will save much code. Maybe something like:

```
if (actualType !== type) {
    return throwOrUndefined(
        `settings: readDevSetting(): setting "${key}": got ${actualType}, expected ${type}`,
        silent)
}
```